### DISCUSSION TITLE
Catch Exception in hook AfterScenario

### DISCUSSION THREAD

**Discussion by carri747** [webdriverio/webdriverio#11567] [Q&A]
Hi everyone and thanks in advance for your help
I've a question about how to manage exceptions in after hooks (after scenarios, after test,...). 
I explain my problem:
I've a situation where I need to restore some test in database after my scenario but if something happen in database and I get a error during this process in the "after scenarios" WDIO has the result "PASSED". This is true because the test was completed fine but I can not identify the error in the after and it's a problem because of the next step wait the data correctly. 
I've a "Try catch" block in the after scenarios and I can see the logs in the catch when error happen but the result still "PASSED", make sense but I don't know if there is any way to identify this or stop the execution or show this error in the report
any help with that?

**Comment by christian-bromann** [comment]
@carri747 I don't understand your issue. You are using try/catch but expect an error to be thrown?

**Comment by carri747** [comment]
I'll try to explain myself a little better. I have this code:
```
Scenario1: My test
Given I do something
When I do something more
Then I check

Scenario2: My test2
Given I do something
When I do something more
Then I check

wdio.conf.ts
afterScenario: async function (world, result) {
try{
// something don't work
}catch{
log.info('error!')
throw new Error(`ERROR!}`);
}
}
```
the situation is:
In the first scenario works fine, I mean all the steps are "passed" and the scenario is "PASSED" but for some reason in the "afterScenario" my code fail in the "try" restoring something that need the follow scenario. Obviously the execution continues and the next scenario fail because the previous one has not restored the data. 
The result is: 

- I can see the error in console. That's fine
- I can NOT see the fail in the scenario1 in my report (allure report) because the scenario is "PASSED"

I don't know if it's my fault or it's something that is not working properly but I would like to see the error in the report and the test with status "FAILED"

I think that now it's better explain 

Thanks for your help @christian-bromann

**Reply by christian-bromann** [reply]
Can you try throwing a [`SevereServiceError`](https://webdriver.io/docs/customservices/#service-error-handling)? By design custom errors in hooks are __not__ causing the testrunner to fail.
